### DISCUSSION TITLE
How to get rid of the space above the footer?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#41126] [Help] [answered]
So I put the footer element at the bottom of _ap.js  and I noticed space between the footer and the rest of the document.

I have looked everywhere to see where is this space created...  is it coming from standard settings when placing a footer?

How could I get rid of this space? 

This is the _app.js file
https://github.com/Elindo586/nextprojectb/blob/main/pages/_app.js

**Comment by icyJoseph** [comment, marked as answer]
Hi,

The problem you have is called `margin collapse`.

You see there's a nested `footer` within your `footer`,  which has `margin-top: 2em`, and that collapses on the upper vertical and spaces out with the rest of content.

- https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Box_Model/Mastering_margin_collapsing
- Here's a rather spicy post, https://mxstbr.com/thoughts/margin/, take it with a grain of salt
- Here's a deep dive into margin collapse https://www.joshwcomeau.com/css/rules-of-margin-collapse/ - highly recommended read

So the solution is to get rid of that margin, which comes from the `main-footer` class. You can find it in `styles/general.css`.

Or shape up the HTML and remove the nested footer :)

**Comment by Elindo586** [comment]
Ah well thanks... it was my fault after all...  I just wanted to make sure there wasn't anything such as "default setting" making this.......    

I was working on another site and found an extra `;`  in the _app.js giving me the same extra space....
